lib/ansible/modules/cloud/memset/memset_memstore_user.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Simon Weald <ansible@simonweald.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: memset_memstore_user
+author: "Simon Weald (@glitchcrab)"
+version_added: "2.9"
+short_description: Manage Memstore users
+notes:
+  - Create/delete and manage the status of Memstore users.
+  - An API key generated via the Memset customer control panel is needed with
+    the following minimum scope - I(memset.user_list), I(memstore.user_info),
+    I(memstore.user_enable), I(memstore.user_disable), I(memstore.user_delete),
+    I(memstore.user_create).
+description:
+    - Create/delete and manage the status of Memstore users.
+options:
+    state:
+        required: true
+        default: present
+        description:
+            - Indicates desired state of resource.
+        choices: [ absent, present ]
+    api_key:
+        required: true
+        description:
+            - The API key obtained from the Memset control panel.
+    username:
+        required: true
+        description:
+            - The name of the user to manage.
+        aliases: [ name ]
+    memstore:
+        required: true
+        description:
+            - The Memstore product the user belongs to.
+    enabled:
+        required: false
+        default: false
+        type: bool
+        description:
+            - Whether the user is enabled or disabled. Defaults to false for security.
+    password:
+        required: false
+        description:
+            - A password for the user. Required when the user is present.
```

Comment:
Does it really have to be required?  Would it make sense to auto generate a random one?

Reply:
We kicked that idea around internally but most customer use-cases (in our experience) were for long-lived passwords. I've no strong opinions either way, but from an ease-of-use perspective I lean towards them providing a password they want to use.